gradio/utils.py

```diff
@@ -689,6 +690,39 @@ def is_update(val):
     return type(val) is dict and "update" in val.get("__type__", "")
 
 
+def get_continuous_fn(fn, every):
+    def continuous_fn(*args):
+        called_before = False
+        while True:
+            output = fn(*args)
+            if called_before:
+                time.sleep(every)
+            else:
+                called_before = True
+            yield output
+
+    return continuous_fn
+
+
+async def cancel_tasks(task_ids: List[str]):
```

Comment:
Refactoring some of the code from the cancels PR so that it's easier to use in other parts of the codebase